mesonbuild/linkers/linkers.py

```diff
@@ -700,14 +700,31 @@ def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
         return (args, rpath_dirs_to_remove)
 
     def get_win_subsystem_args(self, value: str) -> T.List[str]:
-        if 'windows' in value:
-            args = ['--subsystem,windows']
-        elif 'console' in value:
-            args = ['--subsystem,console']
-        else:
-            raise MesonException(f'Only "windows" and "console" are supported for win_subsystem with MinGW, not "{value}".')
+        # Subsystem affects a single entry at the beginning of PE executables.
+        # It is used to verify that an executable is of the expected
+        # application type before attempting to execute it.
```

Comment:
This is not an ideal comment, since we don't much care what it does, especially not as some mingw-specific thing.

Instead we should say something like "mingw only directly supports a couple of the possible PE application types. The raw integer works as an argument as well, and is always accepted, so we manually map the other types here. See LINK for the list of all types".

Reply:
I knew something should be said, but not exactly what as I'm not familiar with the codebase. 
Thanks for the great example, I'll update the comment.
 I'm going to link to the same MSDN article the docs link to, but if you want me to link to somewhere in the meson docs I'll change it.

Reply:
Since the code comment is mainly concerned with "what are these magic values anyway" I would probably go with the wine source code link.